scripts/add-ocp-labels/main.go

```diff
@@ -0,0 +1,634 @@
+	Error string
+}
+
+type Output struct {
+	Errors                       []BundleDir
+	RequiresUpdate               []BundleDir
+	RequiresUpdatePkgFormat      []BundleDir
+	BundleUpdated                []BundleDir
+	BundleUnload                 []BundleDir
+	BundlesConfiguredAccordingly []BundleDir
+	BundlesMigrated              []BundleDir
+	UpdatedUsingServerMode       []BundleDir
+	PkgManifestUsingServerMode   []BundleDir
+	CurrentPath                  string
+}
+
+func main() {
+	createOutput()
+}
+
+func createOutput() {
+	currentPath, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	output := Output{CurrentPath: currentPath}
+	output.updateAllBundles()
+	output.checkIfWeCanLoadTheBundles()
+
+	// Only to add info in the output report
+	output.checkBundlesWhichRequiresUpdateAfterChangesMade()
+	output.checkBundlesUpdateUsingSemver()
+	output.checkPkgUsingSemver()
+
+	output.sort()
+	output.write()
+
+}
+
+func (o *Output) checkPkgUsingSemver() {
+	// Check upgrade graph in 4.9+ of the bundles updated
+	for _, bu := range o.RequiresUpdatePkgFormat {
+		pkgDir := filepath.Clean(filepath.Join(bu.Path, ".."))
+
+		// check if was not added already
+		found := false
+		for _, bh := range o.PkgManifestUsingServerMode {
+			if bh.Path == pkgDir {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+
+		err2 := filepath.Walk(filepath.Join(o.CurrentPath, pkgDir), func(pathPkg string, infoPkg os.FileInfo, err2 error) error {
+			if err2 != nil {
+				return err2
+			}
+			if infoPkg != nil && !infoPkg.IsDir() && strings.Contains(infoPkg.Name(), "ci.yaml") {
+				filePath := filepath.Join(o.CurrentPath, pkgDir, infoPkg.Name())
+				if _, err := os.Stat(filePath); err == nil && !os.IsNotExist(err) {
+					infoPkg, err := ReadFile(filePath)
+					if err != nil {
+						log.Infof("unable to read infoPkg: %s", err)
+						return nil
+					}
+
+					var upgradeGraph UpdateGraph
+					if err := yaml.Unmarshal(infoPkg, &upgradeGraph); err == nil {
+						if upgradeGraph.UpdateGraph == "semver-mode" {
+							o.PkgManifestUsingServerMode = append(o.PkgManifestUsingServerMode, BundleDir{Path: pkgDir})
+							return nil
+						}
+					}
+				}
+				return nil
+			}
+			return nil
+		})
+		if err2 != nil {
+			log.Fatalf("checking server mode", err2)
+		}
+
+	}
+}
+
+func (o *Output) checkBundlesUpdateUsingSemver() {
+	// Check upgrade graph in 4.9+ of the bundles updated
+	for _, bu := range o.BundleUpdated {
+		pkgDir := filepath.Clean(filepath.Join(bu.Path, ".."))
+
+		// check if was not added already
+		found := false
+		for _, bh := range o.UpdatedUsingServerMode {
+			if bh.Path == pkgDir {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+
+		err2 := filepath.Walk(filepath.Join(o.CurrentPath, pkgDir), func(pathPkg string, infoPkg os.FileInfo, err2 error) error {
+			if err2 != nil {
+				return err2
+			}
+			if infoPkg != nil && !infoPkg.IsDir() && strings.Contains(infoPkg.Name(), "ci.yaml") {
```

Comment:
suggestion: Looks similar to the code above. You may want to consider factoring it out.

Reply:
We will remove the script. It is to be used only once. Because of this we only care here about its working as suppose to be.